### ISSUE TITLE
Improve satisfying alert shortages

### ISSUE THREAD

**Issue by bengardner** [year6b7a/item-network-factorio-mod#55]
I played with this a bit more and learned more how Factorio logistic networks work.

The "network" (`LuaLogisticNetwork`) is a collection of "cells" (`LuaLogisticCell`).

The "cells" each have a function `is_in_construction_range(position)` that tells whether there is a cell that can do construction.
It also has a `stationed_construction_robot_count` count.

Items can only be given to a network, not a cell.

It looks like a position should only be contained in one network, but I'm not sure about that. There could be multiple networks that overlap construction areas, but are not linked.

The steps are to find all overlapping nets and then check them for the first cell within construction range that has an available construction robot.
```lua
local nets = entity.surface.find_logistic_networks_by_construction_area(entity.position, "player")
for _, net in ipairs(nets) do
  for _, cell in ipairs(net.cells) do
    if cell.stationed_construction_robot_count > 0 and cell.is_in_construction_range(entity.position) then
       if try_to_send_to_net(net, name, entity) then
         return
       end
    end
  end
end
```
While researching that, I also realized that the player can have a logistic network and the same approach can be used.
```lua
  if player.character ~= nil and player.character.logistic_network ~= nil then
    net = player.character.logistic_network
    for _, cell in ipairs(net.cells) do
    if cell.stationed_construction_robot_count > 0 and cell.is_in_construction_range(entity.position) then
        if try_to_send_to_net(net, name, entity) then
          return
        end
      end
    end
  end
```

Spidertrons could also be used in the same way, but I haven't tried that.

I'm going to refactor the code use be more generic and use stationary logistic networks, player logistic networks and maybe spidertron logistic networks.
I may also have to try all "characters" attached to the player, as some mods use a dummy character in, say, a train car, for logistics.

**Comment by bengardner** [comment]
Looks like I was overthinking this.  `find_logistic_networks_by_construction_area()` does the trick.

**Comment by bengardner** [comment]
Fixed by #56

**Comment by bengardner** [comment]
I should note that using `find_logistic_networks_by_construction_area()` will hit any logistic network. It works with the land-based roboport, the personal roboport and the one in the spidertron.
